### ISSUE TITLE
rc versions aren't compatible with wildcard entries in package.json

### ISSUE THREAD

**Issue by leggomuhgreggo** [DataDog/dd-sdk-reactnative#238]
Hello! 

Not sure if this counts as a bug per se -- more of an edge case I wanted to report for your consideration, and in case others encounter.

**TL;DR:** Because of a nuance with npm's / semver's resolution logic -- if you use a wildcard version for npm packages, in your package.json, it wont resolve "pre-release" versions when installing -- instead rolling back to the last standard release.
```
  "dependencies": {
    "@datadog/mobile-react-native": "*", // <-- won't resolve @1.0.0-rc9 :(
  }
```

I can't imagine this would have much bearing on y'all's prioritization, since it's such an idiosyncratic edge case, but -- for nx monorepos, with RN or Expo projects, that use datadog -- it requires a slightly tiresome manual work around. So I thought I'd let ya know and also share the report in case others encounter this difficulty.

Looking forward to the v1 release! Y'all rock!

Thanks!

##### References:
- https://github.com/npm/npm/issues/8854
- https://semver.npmjs.com/
- https://docs.npmjs.com/about-semantic-versioning

----

### Additional Details

We use both `@datadog/mobile-react-native` and `@datadog/mobile-react-navigation` (currently @1.0.0-rc9) in a few projects within our monorepo.

#### Project-Level vs Root-Level package.json
Our monorepo is set up to observe the ["Single Version Policy"](https://opensource.google/documentation/reference/thirdparty/oneversion) which means that all our npm dependencies are managed from the package.json in the monorepo root. The project-level package are derived from the root package.json, and, by default, projects in the monorepo won't even have a package.json.

#### React Native Resolution Issue
When we first added the React Native projects to our monorepo, we realized that React Native module resolution logic isn't fully consistent with npm mechanics, and required us to add wildcard entries for RN and certain other packages in project-level package.json files, like: 
```
  "react-native": "*",
```

#### Workaround
I wont go into all the fussy nuance of the build reqs -- but, suffice it to say, that normally the wildcard would be able to resolve the version defined in the corresponding library's entry in the root-level package.json -- but sadly rc versions wont take. So we have to manage them manually be specifying the version across all the project-level package.json -- which is a simple, though error prone, process.

**Comment by leggomuhgreggo** [comment]
Closing, since not technically a bug 👍

**Comment by louiszawadzki** [comment]
I @leggomuhgreggo, thanks for letting us know of this issue, it's definitely going to help us push towards 1.0.0 very soon :)
